src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden === false) {
+      victim.health -= 50;
+    };
+
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
What if victim was bitten but it's still alive? Nothing will change, add some type of check on whether you should filter the array. And create a static method for this logic

Reply:
ty for review, check please the new version:)